crates/supervisor/core/src/syncnode/node.rs

```diff
@@ -64,73 +69,33 @@ pub struct ManagedNode {
 
 impl ManagedNode {
     /// Creates a new [`ManagedNode`] with the specified configuration.
-    pub const fn new(config: Arc<ManagedNodeConfig>) -> Self {
-        Self { config, stop_tx: None, task_handle: None }
+    pub fn new(config: Arc<ManagedNodeConfig>) -> Result<Self, ClientError> {
+        Ok(Self { config, ws_client: Mutex::new(None), stop_tx: None, task_handle: None })
     }
 
-    /// Processes a managed event received from the subscription.
-    ///
-    /// Analyzes the event content and takes appropriate actions based on the
-    /// event fields.
-    /// TODO: Call relevant DB functions to update the state
-    pub fn handle_managed_event(event_result: Option<ManagedEvent>) {
-        match event_result {
-            Some(event) => {
-                debug!(target = "managed_node", event = ?event, "Handling ManagedEvent");
-
-                // Process each field of the event if it's present
-                if let Some(reset_id) = &event.reset {
-                    info!(target = "managed_node", reset_id = %reset_id, "Reset event received");
-                    // Handle reset action
-                }
-
-                if let Some(unsafe_block) = &event.unsafe_block {
-                    info!(target = "managed_node", ?unsafe_block, "Unsafe block event received");
-                    // Handle unsafe block
-                }
-
-                if let Some(derived_ref_pair) = &event.derivation_update {
-                    info!(target = "managed_node", ?derived_ref_pair, "Derivation update received");
-                    // Handle derivation update
-                }
-
-                if let Some(derived_ref_pair) = &event.exhaust_l1 {
-                    info!(
-                        target = "managed_node",
-                        ?derived_ref_pair,
-                        "L1 exhausted event received"
-                    );
-                    // Handle L1 exhaustion
-                }
+    /// Returns a reference to the WebSocket client, creating it if it doesn't exist.
+    // todo: support http client as well
+    pub async fn get_ws_client(&self) -> Result<Arc<WsClient>, ClientError> {
+        let mut ws_client_guard = self.ws_client.lock().await;
+        if ws_client_guard.is_none() {
+            let headers = self.create_auth_headers().map_err(|err| {
+                error!(target: "managed_node", %err, "Failed to create auth headers");
+                ClientError::Custom(format!("failed to create auth headers: {}", err))
```

Comment:
```suggestion
                ClientError::Custom(format!("failed to create auth headers: {ett}"))
```
nitpick, this syntax is preferred, pls check if there are other places in this pr diff where this can be fixed too